contracts/tokenbridge/arbitrum/L2AtomicTokenBridgeFactory.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.4;
+
+import {L2GatewayRouter} from "./gateway/L2GatewayRouter.sol";
+import {L2ERC20Gateway} from "./gateway/L2ERC20Gateway.sol";
+import {L2CustomGateway} from "./gateway/L2CustomGateway.sol";
+import {L2WethGateway} from "./gateway/L2WethGateway.sol";
+import {StandardArbERC20} from "./StandardArbERC20.sol";
+import {BeaconProxyFactory} from "../libraries/ClonableBeaconProxy.sol";
+import {aeWETH} from "../libraries/aeWETH.sol";
+import {UpgradeableBeacon} from "@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol";
+import {ProxyAdmin} from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import {TransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+
+/**
+ * @title Layer2 token bridge creator
+ * @notice This contract is used to deploy token bridge on L2 chain.
+ * @dev L1AtomicTokenBridgeCreator shall call `deployL2Contracts` using retryable and that will result in deployment of canonical token bridge contracts.
+ */
+contract L2AtomicTokenBridgeFactory {
+    function deployL2Contracts(
+        bytes memory routerCreationCode,
+        bytes memory standardGatewayCreationCode,
+        bytes memory customGatewayCreationCode,
+        bytes memory wethGatewayCreationCode,
+        bytes memory aeWethCreationCode,
```

Comment:
```suggestion
        bytes calldata routerCreationCode,
        bytes calldata standardGatewayCreationCode,
        bytes calldata customGatewayCreationCode,
        bytes calldata wethGatewayCreationCode,
        bytes calldata aeWethCreationCode,
```
also in all internal functions in this contract

Reply:
this will probably hit stack limit, so you might need to create a new struct that contain all these code and change it into
```suggestion
        CreationCodes calldata cc,
```
not sure worth it lol

Reply:
added new struct, it's also bit cleaner in addition to being more gas efficient